src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java

```diff
@@ -0,0 +1,343 @@
+   * @param path                  the path to the file to be read
+   * @param lock                  whether or not to lock the file. Defaults to false.
+   * @param timeBetweenSizeCheck  wait time between size checks to determine if a file is ready to be read in milliseconds.
+   * @return the file's content and metadata on a {@link FtpFileAttributes} instance
+   * @throws IllegalArgumentException if the file at the given path doesn't exist
+   */
+  protected Result<InputStream, FtpFileAttributes> doRead(@Config FileConnectorConfig config,
+                                                          @Connection FileSystem fileSystem,
+                                                          @DisplayName("File Path") String path,
+                                                          @Optional(defaultValue = "false") @Placement(
+                                                              tab = ADVANCED_TAB) boolean lock,
+                                                          Long timeBetweenSizeCheck) {
+    fileSystem.changeToBaseDir();
+    return fileSystem.read(config, path, lock, timeBetweenSizeCheck);
+  }
+
+  /**
+   * Writes the {@code content} into the file pointed by {@code path}.
+   * <p>
+   * If the directory on which the file is attempting to be written doesn't exist, then the operation will either throw
+   * {@link IllegalArgumentException} or create such folder depending on the value of the {@code createParentDirectory}.
+   * <p>
+   * If the file itself already exists, then the behavior depends on the supplied {@code mode}.
+   * <p>
+   * This operation also supports locking support depending on the value of the {@code lock} argument, but following the same
+   * rules and considerations as described in the read operation.
+   *
+   * @param config                  the {@link FileConnectorConfig} on which the operation is being executed
+   * @param fileSystem              a reference to the host {@link FileSystem}
+   * @param path                    the path of the file to be written
+   * @param content                 the content to be written into the file. Defaults to the current {@link Message} payload
+   * @param createParentDirectories whether or not to attempt creating any parent directories which don't exists.
+   * @param lock                    whether or not to lock the file. Defaults to false
+   * @param mode                    a {@link FileWriteMode}. Defaults to {@code OVERWRITE}
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  protected void doWrite(FileConnectorConfig config, FileSystem fileSystem, String path, InputStream content,
+                         boolean createParentDirectories, boolean lock, FileWriteMode mode) {
+    if (content == null) {
+      throw new IllegalContentException("Cannot write a null content");
+    }
+
+    validatePath(path, "path");
+    fileSystem.changeToBaseDir();
```

Comment:
What if the path is an absolute path? Why do this?

Reply:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.